packages/eas-cli/src/commands/update/republish.ts

```diff
@@ -0,0 +1,373 @@
+    branch: Flags.string({
+      description: 'Branch name to select an update from',
+    }),
+    group: Flags.string({
+      description: 'Update group ID to republish',
+    }),
+    message: Flags.string({
+      description: 'Short message describing the update',
+      required: false,
+    }),
+    platform: Flags.enum({
+      char: 'p',
+      options: [...defaultRepublishPlatforms, 'all'],
+      default: 'all',
+      required: false,
+    }),
+    ...EasNonInteractiveAndJsonFlags,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    const { flags: rawFlags } = await this.parse(UpdateRepublish);
+    const flags = this.sanitizeFlags(rawFlags);
+
+    const {
+      projectConfig: { exp, projectId },
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(UpdateRepublish, {
+      nonInteractive: flags.nonInteractive,
+    });
+
+    if (flags.json) {
+      enableJsonOutput();
+    }
+
+    const existingUpdates = await getOrAskUpdatesAsync(graphqlClient, projectId, flags);
+    const updatesToPublish = existingUpdates.filter(update =>
+      flags.platform.includes(update.platform as Platform)
+    );
+
+    if (existingUpdates.length === 0) {
+      throw new Error(`There are no published updates found`);
+    }
+    if (updatesToPublish.length === 0) {
+      throw new Error(
+        `There are no updates on branch "${
+          existingUpdates[0].branchName
+        }" published for the platform(s) "${rawFlags.platform}" with group ID "${
+          flags.groupId ? flags.groupId : updatesToPublish[0].groupId
+        }". Did you mean to publish a new update instead?`
+      );
+    }
+
+    if (rawFlags.platform === 'all') {
+      Log.withTick(`The republished update will appear only on: ${rawFlags.platform}`);
+    } else {
+      const platformsFromUpdates = updatesToPublish.map(update => update.platform);
+      if (platformsFromUpdates.length < defaultRepublishPlatforms.length) {
+        Log.warn(`You are republishing an update that wasn't published for all platforms.`);
+      }
+
+      Log.withTick(
+        `The republished update will appear on the same platforms it was originally published on: ${platformsFromUpdates.join(
+          ', '
+        )}`
+      );
+    }
+
+    // This command only republishes a single update group
+    // The update group properties are the same for all updates
+    const groupId = updatesToPublish[0].groupId;
+    const runtimeVersion = updatesToPublish[0].runtimeVersion;
+    const branchName = flags.branchName ?? updatesToPublish[0].branchName;
+
+    // Prevent users from republishing updates to a different branch.
+    // When using environment variables, the branch name is not updated, possibly causing unexpected sideeffects.
+    assertBranchNameIsEqualToExistingUpdateBranch(updatesToPublish, branchName, flags);
```

Comment:
```suggestion
```

if we're just going to check that it's equal to the branch name, let's just make the `group` and `branch` flags mutually exclusive (using `exclusive` declaration). Then we can just use the branch name from `updatesToPublish[0].branchName;`.

Reply:
Done!